app/src/org/commcare/android/util/MarkupUtil.java

```diff
@@ -1,19 +1,180 @@
 package org.commcare.android.util;
 
+import in.uncod.android.bypass.Bypass;
 import net.nightwhistler.htmlspanner.HtmlSpanner;
+
+import org.commcare.dalvik.application.CommCareApp;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.dalvik.preferences.DeveloperPreferences;
+import org.javarosa.core.services.locale.Localization;
+
+import android.content.Context;
+import android.text.Html;
 import android.text.Spannable;
+import android.text.SpannableString;
 
 public class MarkupUtil {
```

Comment:
One high level comment:

We're going to have to replace ~Every call to

Localization.get()

with something from this class, and 

MarkupUtil.localizeGenerateMarkdown(context, text)

Might get pretty brutal. 

Two recommendations:

1) Let's come up with a super concise/explanatory-sounding way to write this like

AndroidText.localized(context, string)

Or something else that is shorter/looks intuitive.

B) We can work around the need for the context if we make each CommCareActivity (which is a context) forward on these method signatures without it. IE:

activity.localize(String)

which would likely make life easier long-term and make coding less of a chore.